code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -132,23 +188,49 @@
 			user.visible_message("<span class='danger'>The shotgun goes off!</span>", "<span class='danger'>The shotgun goes off in your face!</span>")
 			return
 		if(do_after(user, src, 30))	//SHIT IS STEALTHY EYYYYY
-			icon_state = "sawnshotgun"
-			w_class = W_CLASS_MEDIUM
-			item_state = "sawnshotgun"
-			slot_flags &= ~SLOT_BACK	//you can't sling it on your back
-			slot_flags |= SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
-			name = "sawn-off shotgun"
-			desc = "Omar's coming!"
+			qdel(src)
 			to_chat(user, "<span class='warning'>You shorten the barrel of \the [src]!</span>")
-			if(istype(user, /mob/living/carbon/human) && src.loc == user)
-				var/mob/living/carbon/human/H = user
-				H.update_inv_hands()
+			var/sawn = new /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff
+			user.put_in_hands(sawn)
+	if(!broke)
+		return
 
 /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff
 	name = "sawn-off shotgun"
 	desc = "Omar's coming!"
 	icon_state = "sawnshotgun"
 	item_state = "sawnshotgun"
 	w_class = W_CLASS_MEDIUM
+	flags = FPRINT
 	slot_flags = SLOT_BELT
-	ammo_type = "/obj/item/ammo_casing/shotgun/buckshot"
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawnoff/attack_self(mob/living/user as mob)
```

Comment:
Make a variable whether the thing can be wielded, and then override it for the sawn off.